src/hifz/dataserver.py

```diff
@@ -18,7 +19,10 @@ def read_entries(self, file_path: str) -> list[Card]:
 
         try:
             with Path(file_path).open() as f:
-                entries = csv.DictReader(f)
+                if file_path.lower.endswith(".csv"):
+                    entries = csv.DictReader(f)
+                elif file_path.endswith(".json"):
+                    entries = json.load(f)
                 cards = [Card(e["front"], e["back"]) for e in entries]
         except FileNotFoundError as err:
             error_message = f"Error: The file at path '{file_path}' was not found."
```

Comment:
Remember to raise the error.

Reply:
This was my bad, I couldn't see the raise in the GitHub ui but upon closer inspection it was there all along. Good work!